pkg/cluster/sync.go

```diff
@@ -310,6 +336,19 @@ func (c *Cluster) syncStatefulSet() error {
 				}
 			}
 		}
+
+		if !podsRollingUpdateRequired {
```

Comment:
```suggestion
		if c.OpConfig.EnableLazySpiloUpgrade && !podsRollingUpdateRequired {
```

I would move the check for lazy update here instead of doing it within `mustUpdatePodsAfterLazyUpdate

Reply:
I wonder if this part here is really necessary? When lazy update is active it would create another API call for every StatefulSet. Wouldn't the next sync fix outdated images?

Reply:
> I would move the check for lazy update here instead of doing it within mustUpdatePodsAfterLazyUpdate

Changed. Note the condition is `!c.OpConfig.EnableLazySpiloUpgrade` , that is,  we only need to do it when  lazy update is disabled.

>  Wouldn't the next sync fix outdated images?

No. Sync() relies on stateful set spec to figure out if rolling update is needed. With the lazy upgrade, stateful set itself will be up-to-date (pod template will contain the most recent image),  but some of the pods may still be running old images just because they were not yet restarted. Operator will look in the stateful set spec, see the most recent image and happily skip the rolling upgrade for this cluster even though some pods run outdated images.